x/mongo/driver/compression.go

```diff
@@ -25,6 +25,32 @@ type CompressionOpts struct {
 	UncompressedSize int32
 }
 
+func calcZstdWindowSize(n int, l zstd.EncoderLevel) int {
+	windowSize := zstd.MinWindowSize
+	if n > windowSize {
```

Comment:
What are your thoughts on removing this branch altogether?  One edge case I'm thinking of is if this routine is called with an input size of 0, i.e. this test case:

```
{
	inputSize:  0,
	level:      zstd.EncoderLevelFromZstd(wiremessage.DefaultZstdLevel),
	windowSize: 1,
},
```
Perhaps I'm not thinking about it correctly, but I would expect the window size to be 1 since that is the [smallest possible window size](https://github.com/klauspost/compress/blob/v1.15.6/zstd/framedec.go#L37).  Removing the `if` block fixes this problem with this block:

```go
for windowSize/2 > n {
	windowSize /= 2
}
```

Reply:
The smallest possible window size is 1024 (1 << 10). I did not dig very deep to verify if we still need a minimal window size for 0 input. However, it will not be wrong to keep it. Also, I keep the const of `MinWindowSize` as a safeguard to prevent any future changes in the third-part package.

Otherwise, I agree with @matthewdale that an early return for anything smaller than the `MinWindowSize` may be more readable.